exchanges/huobi/huobi_test.go

```diff
@@ -3010,3 +3012,52 @@ func TestGetCurrencyTradeURL(t *testing.T) {
 		}
 	}
 }
+
+func TestGenerateSubscriptions(t *testing.T) {
+	t.Parallel()
+
+	h := new(HUOBI)
+	require.NoError(t, testexch.Setup(h), "Test instance Setup must not error")
+
+	h.Websocket.SetCanUseAuthenticatedEndpoints(true)
+	subs, err := h.generateSubscriptions()
+	require.NoError(t, err, "generateSubscriptions must not error")
+	exp := subscription.List{}
+	for _, s := range h.Features.Subscriptions {
+		for _, a := range h.GetAssetTypes(true) {
+			if s.Asset != asset.All && s.Asset != a {
+				continue
+			}
+			pairs, err := h.GetEnabledPairs(a)
+			require.NoErrorf(t, err, "GetEnabledPairs %s must not error", a)
+			pairs = common.SortStrings(pairs).Format(currency.PairFormat{Uppercase: false, Delimiter: ""})
+			s := s.Clone() //nolint:govet // Intentional lexical scope shadow
+			s.Asset = a
+			for i, p := range pairs {
+				s := s.Clone() //nolint:govet // Intentional lexical scope shadow
+				s.QualifiedChannel = channelName(s, p)
+				switch s.Channel {
+				case subscription.OrderbookChannel:
+					s.QualifiedChannel += ".step0"
+				case subscription.CandlesChannel:
+					s.QualifiedChannel += ".1min"
+				}
+				s.Pairs = pairs[i : i+1]
+				exp = append(exp, s)
+			}
+		}
+	}
+	testsubs.EqualLists(t, exp, subs)
+}
+
+func TestSubscribe(t *testing.T) {
+	t.Parallel()
+	h := new(HUOBI)
+	require.NoError(t, testexch.Setup(h), "Test instance Setup must not error")
+	subs, err := h.Features.Subscriptions.ExpandTemplates(h)
+	require.NoError(t, err, "ExpandTemplates must not error")
+	h.Features.Subscriptions = subscription.List{}
+	testexch.SetupWs(t, h)
```

Comment:
```
    huobi_test.go:3060: Websocket not enabled
--- SKIP: TestSubscribe (0.01s)
```
🤷 There seems to be no Github Actions running on this pr too

Anyway, I request a mock subscribe test here too.

Reply:
I implemented mocks for authed subs, but it also meant implementing V2 auth, and the rabbit hole became entirely too much.
So I've enabled websocket and commented out auth endpoints in the translation table.

The [v2 implementation](https://github.com/gbjk/gocryptotrader/pull/new/feature/huobi_v2) will be ready very soon, and that commenting can go away.

I've deliberately left the authed subs in config so that we don't need to do a config upgrade to enable v2.
Anyone using authed WS is broken already and won't be any more broken as a result of merging this PR.

Fixed 86b8b4c0

Reply:
Well I appreciate that v2 WS is on its way 🎉 
With all of that and no auth in this PR, I will delay demanding auth sub coverage to that future PR.